p2p/enode/urlv4.go

```diff
@@ -103,6 +103,37 @@ func NewV4(pubkey *ecdsa.PublicKey, ip net.IP, tcp, udp, raftPort int) *Node {
 	return n
 }
 
+// Quorum
```

Comment:
We could break out the shared logic between `NewV4` and `NewV4Hostname` into a private function `newV4` this would make it easier to see there is no change in the logic except for hostname vs IP type. 
```
// NewV4 creates a node from discovery v4 node information. The record
// contained in the node has a zero-length signature.
func NewV4(pubkey *ecdsa.PublicKey, ip net.IP, tcp, udp, raftPort int) *Node {
	var r enr.Record
	if ip != nil {
		r.Set(enr.IP(ip))
	}
        if raftPort != 0 { // Quorum specific (TODO: can we remove this here? and from the function sig?)
		r.Set(enr.RaftPort(raftPort))
	}
	return newV4(pubkey, r, tcp, udp, raftPort)
}
// broken out from `func NewV4` (above) same in upstream go-ethereum, but taken out
// to avoid code duplication b/t NewV4 and NewV4Hostname
func newV4(pubkey *ecdsa.PublicKey, r enr.Record, tcp, udp, raftPort int) *Node {
     if udp != 0 {
       r.Set(enr.UDP(udp))
     }
     if tcp != 0 {
       r.Set(enr.TCP(tcp))
     }

     if raftPort != 0 { // Quorum
       r.Set(enr.RaftPort(raftPort))
     }

     signV4Compat(&r, pubkey)
     n, err := New(v4CompatID{}, &r)
     if err != nil {
       panic(err)
     }
     return n
}
// Quorum

// NewV4Hostname creates a node from discovery v4 node information. The record
// contained in the node has a zero-length signature. It sets the hostname of
// the node instead of the IP address.
func NewV4Hostname(pubkey *ecdsa.PublicKey, hostname string, tcp, udp, raftPort int) *Node {
	var r enr.Record
	if hostname != "" {
		r.Set(enr.Hostname(hostname))
	}
	if raftPort != 0 {
		r.Set(enr.RaftPort(raftPort))
	}
	return newV4(pubkey, r, tcp, udp, raftPort)
}

// End-Quorum
```
Additionally, it seems the `raftPort` is only set in `parseComplete` so the signature for `NewV4` could lose the raftPort `func NewV4(pubkey *ecdsa.PublicKey, ip net.IP, tcp, udp int) *Node` and thus match upstream (https://github.com/ethereum/go-ethereum/blob/master/p2p/enode/urlv4.go#L85), and the Quorum specific `raftport` could be set only in the Quorum specific `NewV4Hostname(pubkey *ecdsa.PublicKey, hostname string, tcp, udp, raftPort int)` before calling `func newV4(pubkey *ecdsa.PublicKey, r enr.Record, tcp, udp int) *Node ` (TODO: check me on this).

Reply:
I made this change, although I originally kept them separate to avoid changing upstream code (only adding onto). As far as the raft port is concerned, you may be right, but I don't plan on messing around with that as it doesn't pertain to this change.